docs/2022-07-18-JSON-RPC-Server-API.md

```diff
@@ -190,6 +193,36 @@ If `"reason": "cut-point-rule"`, the `next-states` field contains the next state
   }
 }
 ```
+If `"reason": "timeout"`, an additional `next-states` field contains all following states.
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": {
+    "state": {
+      "term": {"format":"KORE", "version":1, "term":{}},
+      "predicate": {"format":"KORE", "version":1, "term":{}},
+      "substitution": {"format":"KORE", "version":1, "term":{}},
+    },
+    "depth": 2,
+    "reason": "timeout",
+    "rule": "rule1",
+    "next-states": [
+      {
+        "term": {"format": "KORE", "version": 1, "term": {}},
+        "predicate": {"format":"KORE", "version":1, "term":{}},
```

Comment:
Sorry @Sereja313 , I don't understand how there can be "next states" if the computation timed out. 
If we have finished computing a single next state, we can proceed, if we have more than one we can return `branching`. Am I misunderstanding something?